download_pod_logs/model.py

```diff
@@ -216,15 +216,35 @@ def _write_log(kubectl: KubectlInterface, pod: KubernetesResource, tail: int, ou
                 output_file.writelines(("#" * 50) + "\n\n")
 
                 # call kubectl to get the log for this container
-                try:
-                    log: List[AnyStr] = kubectl.logs(pod_name=f"{pod.get_name()} {container_status.get_name()}",
-                                                     all_containers=False, prefix=False, tail=tail)
 
+                log: List[AnyStr] = list()
+                try:
+                    log: List[AnyStr] = kubectl.logs(pod_name=pod.get_name(),
+                                                     container_name=container_status.get_name(),
+                                                     prefix=False, tail=tail,
+                                                     ignore_errors=False, ignore_warnings=False)
                 except CalledProcessError:
-                    err_msg = (f"ERROR: A log could not be retrieved for the container [{container_status.get_name()}] "
-                               f"in pod [{pod.get_name()}] in namespace [{kubectl.get_namespace()}]")
-                    log: List[AnyStr] = [err_msg]
-                    err_info.append((container_status.get_name(), pod.get_name()))
+                    try:
+                        log: List[AnyStr] = kubectl.logs(pod_name=pod.get_name(),
```

Comment:
I'm not sure I follow what this call is doing. What is this getting that couldn't be retrieved from the CalledProcessError that was raised?

Reply:
The logic was used to retrieve log from initContainers.

Reply:
code updated.